src/aiida/orm/nodes/process/process.py

```diff
@@ -251,7 +259,16 @@ def logger(self):
         """
         from aiida.orm.utils.log import create_logger_adapter
 
-        return create_logger_adapter(self._logger, self)
+        # If the node is not yet stored, there is no point in creating the logger adapter yet as the ``DbLogHandler``
+        # it configures only is triggered for stored nodes otherwise it cannot link the log message to the node.
+        if not self.pk:
+            return self._logger
+
+        # First time the property is called after the node is stored, create the logger adapter
+        if self._logger_adapter is None:
```

Comment:
Although it works, how about we change this condition to:

```suggestion
        if not hasattr(self, '_logger_adapter'):
```
Then, there is no need to define a class variable.

it seems `if self._logger_adapter is None:` first, checks for instance variables; if they cannot be found, it looks for class variables. In the end, it sets instance variables.